exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/DataCollector.java

```diff
@@ -28,4 +29,16 @@
   public RawBatchBuffer[] getBuffers();
   public int getTotalIncomingFragments();
   public void close() throws Exception;
+  /**
+   * Allows an operator (e.g., receiver) to attach its buffer allocator to this Data Collector so that
+   * incoming batches are owned by the operator instead of the parent (fragment). Note
+   * that this operation is optional; the fragment allocator is used in case no operator invokes this method.
+   *
+   * @param oprAllocator operator buffer allocator
+   */
+  void setOprAllocator(BufferAllocator oprAllocator);
```

Comment:
I am not sure that `Opr` qualification is necessary. There is no assumption that it is an operator. It can be any `BufferAllocator`.

Reply:
Agreed. What about get/setOwningAllocator?

Reply:
IMO, get/setAllocator is good enough.